### PR TITLE
Deploy kubernetes-external-secrets to aaa cluster

### PR DESCRIPTION
**Pull request by ameukam**
Use Kubernetes External Secrets for auto sync secrets from GCP Secrets
Manager.
Initial discussion started here : https://docs.google.com/document/d/1Jo8dyNiJ1fnYh30M_6Gq2XDU9B3bGFmkD2QokyKYwEg/.

Signed-off-by: Arnaud Meukam <ameukam@gmail.com>

### CODE REVIEW

File: kubernetes-external-secrets/00-namespace.yaml
Comment (spiffxp) [inline kubernetes-external-secrets/00-namespace.yaml:4]: We should use ensure-namespace for this and make sure there's a group setup too. Make cluster-admins members of the group

File: infra/gcp/ensure-main-project.sh
Comment (spiffxp) [inline infra/gcp/ensure-main-project.sh:195]: Note to self: this will need a redo or a rebase once infra script fix PR lands

**Review by spiffxp — CHANGES_REQUESTED** [review]

File: infra/gcp/ensure-main-project.sh
Comment (ameukam) [reply infra/gcp/ensure-main-project.sh]: @spiffxp I switched to `ensure_project_role_binding`. Is that enough ?

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm
/hold
Remove /hold when ready to deploy

**Pull request merged**
